api/propagation/propagator.go

```diff
@@ -18,22 +18,47 @@ import (
 	"context"
 
 	"go.opentelemetry.io/api/core"
+	dctx "go.opentelemetry.io/api/distributedcontext"
 )
 
 // TextFormatPropagator is an interface that specifies methods to inject and extract SpanContext
-// into/from a carrier using Supplier interface.
+// and distributed context into/from a carrier using Supplier interface.
 // For example, HTTP Trace Context propagator would encode SpanContext into W3C Trace
 // Context Header and set the header into HttpRequest.
 type TextFormatPropagator interface {
 	// Inject method retrieves current SpanContext from the ctx, encodes it into propagator
 	// specific format and then injects the encoded SpanContext using supplier into a carrier
-	// associated with the supplier.
-	Inject(ctx context.Context, supplier Supplier)
+	// associated with the supplier. It also takes a correlationCtx whose values will be
+	// injected into a carrier using the supplier.
+	Inject(ctx context.Context, correlationCtx dctx.Map, supplier Supplier)
 
 	// Extract method retrieves encoded SpanContext using supplier from the associated carrier.
-	// It decodes the SpanContext and returns it. If no SpanContext was retrieved OR
-	// if the retrieved SpanContext is invalid then an empty SpanContext is returned.
-	Extract(ctx context.Context, supplier Supplier) core.SpanContext
+	// It decodes the SpanContext and returns it and a dctx of correlated context.
+	// If no SpanContext was retrieved OR if the retrieved SpanContext is invalid then
+	// an empty SpanContext is returned.
+	Extract(ctx context.Context, supplier Supplier) (core.SpanContext, dctx.Map)
+
+	// GetAllKeys returns all the keys that this propagator injects/extracts into/from a
+	// carrier. The use cases for this are
+	// * allow pre-allocation of fields, especially in systems like gRPC Metadata
+	// * allow a single-pass over an iterator (ex OpenTracing has no getter in TextMap)
+	GetAllKeys() []string
+}
+
+// TextFormatBaggagePropagator is an interface that specifies methods to inject and extract
+// DistributedContext into/from a carrier using Supplier interface.
+// For example, HTTP Correlation Context propagator would encode DistributedContext into W3C Trace
+// Correlation Context Header and set the header into HttpRequest.
+type TextFormatCorrelationContextPropagator interface {
```

Comment:
Is this a Baggage or CorrelationContext? Either way it should be removed. If it is baggage then we want to hold on to adding this until we have clarity on whether to support baggage as part of open-telemetery or an independent contrib package. If it is correlation-context (for now distributedContext) then it is already part of trace-context propagator.

Reply:
Oops, I didn't realize this didn't get deleted. This is unused and should be removed.